dags/intacct_historical.py

```diff
@@ -64,21 +65,23 @@
     )
 
     for location, connector_id in list_of_connectors.items():
-
-        fivetran_sync_start = FivetranOperator(
-                task_id='intacct-task-{}'.format(location),
+        with TaskGroup(f'intacct-{location}', prefix_group_id=False):
+            fivetran_sync_start = FivetranOperator(
+                task_id=f'intacct-task-{location}',
                 fivetran_conn_id='fivetran',
-                connector_id='{}'.format(connector_id)
-        )
-
-        fivetran_sync_wait = FivetranSensor(
-                task_id='intacct-sensor-{}'.format(location),
+                connector_id=connector_id,
+            )
+
+            # It's best if the sensor starts before the Fivetran sync is triggered to avoid any
```

Comment:
But if there is no active sync would this just immediately be marked as successful?

Reply:
No, when the sensor starts running it looks at the last sync that happened, then sits there waiting for another sync to happen.  The sensor only completes when a sync happens during the time the sensor is running.

Reply:
hmm, maybe it would be worth contributing to the community and adding an ability to check if the most recent run in Fivetran occurred within x amount of time of time for cases like this.

Reply:
https://github.com/fivetran/airflow-provider-fivetran/blob/c65bb58130951a90bdcefea63880d2d609f61034/fivetran_provider/hooks/fivetran.py#L284-L290

Looks like somewhere around here we could add a date if after which a success occurred we just return True.

Reply:
But thanks for raising this, let's hope our sync takes longer than it takes for Airflow to schedule the sensor task.

Reply:
> hmm, maybe it would be worth contributing to the community and adding an ability to check if the most recent run in Fivetran occurred within x amount of time of time for cases like this.

> https://github.com/fivetran/airflow-provider-fivetran/blob/c65bb58130951a90bdcefea63880d2d609f61034/fivetran_provider/hooks/fivetran.py#L284-L290
> 
> Looks like somewhere around here we could add a date if after which a success occurred we just return True.

A similar idea I had was using XCom to pass the sync start timestamp to the sensor.

Or another idea would be to have a single custom operator that starts the sync and waits for it to complete.  A big benefit of that would be making retries of failed syncs straightforward.  As it is, failed syncs won't get retried unless we find a way to have the sensor task retries also trigger the associated sync task to retry (seems possible with using `on_retry_callback` and/or sub-DAGs, but complicated).

However, for this PR I'm trying to keep things simple and mitigate the problems we're seeing without committing to more custom Fivetran Airflow operator/sensor code (partly because I think it'd be good if we can use Fivetran's provided packages rather than our customized copy of it).